### PR TITLE
create wallet account

### PR DESCRIPTION
**Pull request by J-Son89**
fixes: 
https://github.com/status-im/status-mobile/issues/17061

This pr implements the ability to create an account in the new wallet ui using the default keypair and wallet-root-address.

 To test this branch I worked with @mmilad75's pr: https://github.com/status-im/status-mobile/pull/17447 - see the video attached
  
 

https://github.com/status-im/status-mobile/assets/22799766/e44b14e6-9675-4af7-92f8-b01bd374a721

There are some follow ups to this work ->
improve mechanism to derive path -> should be based upon the derivation path of last derived path sharing the same key-uid. 
Navigation currently goes to the wrong place after creating an account. there is some screens that need to be added for animation while it generates the account. 

Testing ->
To test this feature it generates a new wallet account in the new UI. To see that the new account is generated in the UI you can create an account and then log out and in again and check the old wallet UI to see that there is a new account in that list. Also if the create account request is successful the page will navigate to the default account page.
It's also important to create a new account to test this feature. From a QA perspective it might also be okay to ignore harshly testing this feature until some other work is in place so we can test these features in full, e.g the real list of users accounts displayed so we can see accounts are generated and they have the correct name, emoji etc 👍 

 this touches off the basic auth mechanism which is also used on the syncing page 👍

### CODE REVIEW

File: src/status_im2/common/standard_authentication/standard_auth/view.cljs [hunk 1]
```diff
@@ -40,11 +40,13 @@
          (rf/dispatch [:show-bottom-sheet
                        {:on-close on-close
                         :theme    theme
-                        :shell?   blur?
+                        :shell?   false
```
Comment (J-Son89) [inline src/status_im2/common/standard_authentication/standard_auth/view.cljs]: I need to swap this back 😓

File: src/status_im2/contexts/profile/login/events.cljs [hunk 2]
```diff
@@ -75,6 +75,7 @@
               {:db (assoc db
                           :chats/loading?           true
                           :networks/current-network current-network
+                          :wallet-2/tokens-loading? true
```
Comment (J-Son89) [inline src/status_im2/contexts/profile/login/events.cljs]: I need to revert this, related to @mmilad75's work

File: src/status_im2/contexts/profile/login/events.cljs [hunk 3]
```diff
@@ -190,14 +191,15 @@
...
+                      :on-success #(rf/dispatch [:profile.login/verified-database-password % cb])
                       :on-error   #(log/error "accounts_verifyPassword error" %)}]}))
 
 (rf/defn verify-database-password-success
   {:events [:profile.login/verified-database-password]}
-  [{:keys [db] :as cofx} valid?]
+  [{:keys [db] :as cofx} valid? callback]
   (if valid?
     (do
+      (callback)
```
Comment (J-Son89) [inline src/status_im2/contexts/profile/login/events.cljs]: callback should only fire if password is valid 👍

File: src/status_im2/contexts/wallet/common/temp.cljs [hunk 4]
```diff
@@ -119,28 +118,17 @@
...
-                        :icon-left   :i/placeholder
-                        :alignment   :flex-start}
-    :description       :text
-    :description-props {:text (string/replace constants/path-default-wallet #"/" " / ")}}])
+  [name derivation-path]
+  [{:title        (keypair-string name)
+    :button-props {:title (i18n/label :t/edit)}
+    :left-icon    :i/placeholder
+    :description :text
+    :description-props {:text "On device"}}
```
Comment (J-Son89) [inline src/status_im2/contexts/wallet/common/temp.cljs]: will make it an i18n label

File: src/status_im2/contexts/wallet/events.cljs [hunk 5]
```diff
@@ -0,0 +1,40 @@
+(ns status-im2.contexts.wallet.events
+  (:require [utils.re-frame :as rf]
+            [taoensso.timbre :as log]
+            [utils.security.core :as security]
+            [native-module.core :as native-module]))
+
```
Comment (J-Son89) [inline src/status_im2/contexts/wallet/events.cljs:17]: @ilmotta would appreciate a review on these events 👍

Comment (J-Son89) [reply src/status_im2/contexts/wallet/events.cljs:17]: also do we have a way to test these rpc requests? 🤔 -  I can add them here if there's a good example?
cc @cammellos

File: src/utils/re_frame.cljs [hunk 6]
```diff
@@ -78,3 +78,5 @@
 (def sub (comp deref re-frame/subscribe))
 
 (def dispatch re-frame/dispatch)
+
+(def reg-event-fx re-frame/reg-event-fx)
```
Comment (J-Son89) [inline src/utils/re_frame.cljs:87]: @ilmotta added this so we still work through `rf/...` wdyt?

Comment (ilmotta) [reply src/utils/re_frame.cljs:87]: Seems reasonable to me @J-Son89

Comment (J-Son89) [reply src/utils/re_frame.cljs:87]: I'll make a follow up pr to adjust the other uses so 👍

File: src/status_im2/contexts/wallet/events.cljs (hunk 5 above)
Comment (ilmotta) [reply src/status_im2/contexts/wallet/events.cljs:17]: > @ilmotta would appreciate a review on these events 

Will do :)

> also do we have a way to test these rpc requests?

The integration tests can make actual JSON RPC calls. If you would like to be sure more complex flows are correct I believe that's the place. If you want to venture in those tests, it would be good to write them in a separate file than `status-im.integration-test`, that one is crowded.

Comment (J-Son89) [reply src/status_im2/contexts/wallet/events.cljs:17]: that filename sounds like it could be improved upon to me anyway. Will put them in wallet specific test files. 
sounds great. thanks 👍

File: src/status_im2/contexts/profile/login/events.cljs (hunk 3 above)
Comment (smohamedjavid) [reply src/status_im2/contexts/profile/login/events.cljs]: It's good to check if the callback is available and it's a function.
```suggestion
 (when (fn? callback)
      (callback))
```

File: src/status_im2/contexts/wallet/create_account/view.cljs [hunk 7]
```diff
@@ -18,7 +20,11 @@
         bottom               (safe-area/get-bottom)
         account-color        (reagent/atom :blue)
         emoji                (reagent/atom diamond-emoji)
+        number-of-accounts  (count (rf/sub [:profile/wallet-accounts]))
+        account-name         (reagent/atom (i18n/label :t/default-account-name {:number (inc number-of-accounts)}))
+        derivation-path (reagent/atom (wallet-utils/get-derivation-path number-of-accounts))
         {:keys [public-key]} (rf/sub [:profile/profile])
+        on-change-text (fn [v] (reset! account-name v))
```
Comment (smohamedjavid) [inline src/status_im2/contexts/wallet/create_account/view.cljs]: We could use an anonymous function here
```suggestion
        on-change-text #(reset! account-name %)
```

File: src/status_im2/contexts/wallet/events.cljs [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+(rf/reg-event-fx :wallet-2/create-derived-addresses
+                 (fn [{:keys [db]} [password {:keys [path]} on-success]]
+                   (let [{:keys [wallet-root-address]} (:profile/profile db)
+                         sha3-pwd   (native-module/sha3 (str (security/safe-unmask-data password)))]
+                     {:json-rpc/call [{:method     "wallet_getDerivedAddresses"
+                                       :params     [sha3-pwd (clj->js wallet-root-address) (clj->js [path])]
+                                       :on-success on-success
+                                       :on-error   #(log/info "failed to derive address " %)}]})))
+
```
Comment (smohamedjavid) [inline src/status_im2/contexts/wallet/events.cljs]: Moving the event name to the next line would be more readable.

```suggestion
(rf/reg-event-fx 
:wallet-2/create-derived-addresses
                 (fn [{:keys [db]} [password {:keys [path]} on-success]]
                   (let [{:keys [wallet-root-address]} (:profile/profile db)
                         sha3-pwd   (native-module/sha3 (str (security/safe-unmask-data password)))]
                     {:json-rpc/call [{:method     "wallet_getDerivedAddresses"
                                       :params     [sha3-pwd (clj->js wallet-root-address) (clj->js [path])]
                                       :on-success on-success
                                       :on-error   #(log/info "failed to derive address " %)}]})))

```

File: src/status_im2/contexts/wallet/utils.cljs [hunk 9]
```diff
@@ -0,0 +1,6 @@
+(ns status-im2.contexts.wallet.utils
+  (:require [status-im2.constants :as constants]))
+
+(defn get-derivation-path [number-of-accounts]
+  
+  (str constants/path-wallet-root "/" number-of-accounts))
```
Comment (smohamedjavid) [inline src/status_im2/contexts/wallet/utils.cljs]:
```suggestion
(defn get-derivation-path 
  [number-of-accounts]
  (str constants/path-wallet-root "/" number-of-accounts))
```

File: src/status_im2/contexts/wallet/events.cljs [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+                                       :params     [sha3-pwd account-config]
+                                       :on-success #(rf/dispatch [:navigate-to :wallet-accounts])
+                                       :on-error   #(log/info "failed to create account " %)}]})))
+
+(rf/reg-event-fx
+ :wallet-2/derive-address-and-add-account
+ (fn [_ [password  account-details]]
+   (let [on-success  (fn [derived-adress-details]
+                       (rf/dispatch [:wallet-2/add-account password account-details (first derived-adress-details)]))]
+     (rf/dispatch [:wallet-2/create-derived-addresses password account-details on-success]))))
```
Comment (ilmotta) [inline src/status_im2/contexts/wallet/events.cljs]: @J-Son89, in re-frame, event handlers should never directly call `re-frame.core/dispatch`, only effects should do that (`:json-rpc/call` is one example that does this behind the scenes). The event handler should be pure and always either return nil or a map of effects.

I updated the guidelines recently:

> Register events with `re-frame.core/reg-event-fx` and follow [re-frame's best
practice](https://github.com/day8/re-frame/blob/39adca93673f334dc751ee2d99d340b51a9cc6db/docs/FAQs/BestPractice.md#use-the-fx-effect) so use only `:db` and `:fx` effects.
>
> -- https://github.com/status-im/status-mobile/blob/6e897f0ea6113f8242d4180f52eae6711a9c9938/doc/new-guidelines.md#registering-event-handlers

So here the code would be:

```clojure
(rf/reg-event-fx :wallet-2/derive-address-and-add-account
 (fn [_ [password account-details]]
   (let [on-success (fn [derived-adress-details]
                      (rf/dispatch [:wallet-2/add-account password account-details
                                    (first derived-adress-details)]))]
     {:fx [[:dispatch [:wallet-2/create-derived-addresses password account-details on-success]]]})))
```

File: src/status_im2/contexts/wallet/events.cljs [hunk 11]
```diff
@@ -0,0 +1,40 @@
+(ns status-im2.contexts.wallet.events
+  (:require [utils.re-frame :as rf]
+            [taoensso.timbre :as log]
+            [utils.security.core :as security]
+            [native-module.core :as native-module]))
+
+(rf/reg-event-fx
```
Comment (ilmotta) [inline src/status_im2/contexts/wallet/events.cljs]: The more common way events are formatted in re-frame is to keep the event ID in the same line as `reg-event-fx`. I recently updated zprint to take this into account so that the args after the event ID are aligned like `re-frame.core/reg-sub` (see https://github.com/status-im/status-mobile/blob/6e897f0ea6113f8242d4180f52eae6711a9c9938/.zprintrc#L34)

File: src/status_im2/contexts/wallet/events.cljs [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+                                                  :wallet false
+                                                  :chat false
+                                                  :type :generated
+                                                  :name account-name
+                                                  :emoji emoji
+                                                  :path path
+                                                  :address  address
+                                                  :public-key public-key
+                                                  :colorID color})]
+                     {:json-rpc/call [{:method     "accounts_addAccount"
```
Comment (ilmotta) [inline src/status_im2/contexts/wallet/events.cljs]: To follow the new guidelines on events, the code is slightly different, using `:fx`. `:fx` is the only effect we need to describe all other effects (ordered) in the event handler.

```clojure
(rf/reg-event-fx :wallet-2/add-account
 (fn [{:keys [db]} [password {:keys [emoji account-name color]} {:keys [public-key address path]}]]
   (let [key-uid        (get-in db [:profile/profile :key-uid])
         sha3-pwd       (native-module/sha3 (security/safe-unmask-data password))
         account-config (clj->js {:key-uid    key-uid
                                  :wallet     false
                                  :chat       false
                                  :type       :generated
                                  :name       account-name
                                  :emoji      emoji
                                  :path       path
                                  :address    address
                                  :public-key public-key
                                  :colorID    color})]
     {:fx [[:json-rpc/call
            [{:method     "accounts_addAccount"
              :params     [sha3-pwd account-config]
              :on-success #(rf/dispatch [:navigate-to :wallet-accounts])
              :on-error   #(log/info "failed to create account " %)}]]]})))
```

File: src/status_im2/contexts/wallet/events.cljs [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+                                       :params     [sha3-pwd account-config]
+                                       :on-success #(rf/dispatch [:navigate-to :wallet-accounts])
```
Comment (ilmotta) [inline src/status_im2/contexts/wallet/events.cljs]: FYI: `:on-success` can be an _event vector_ too, leaving the dispatch responsibility to the `:json-rpc/call` effect and being more friendly for unit testing (because of equality checks).

```clojure
:on-success [:navigate-to :wallet-accounts]
```

File: src/status_im2/contexts/wallet/events.cljs [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+                         sha3-pwd   (native-module/sha3 (str (security/safe-unmask-data password)))]
+                     {:json-rpc/call [{:method     "wallet_getDerivedAddresses"
+                                       :params     [sha3-pwd (clj->js wallet-root-address) (clj->js [path])]
+                                       :on-success on-success
+                                       :on-error   #(log/info "failed to derive address " %)}]})))
+(rf/reg-event-fx :wallet-2/add-account
+                 (fn [{:keys [db]} [password {:keys [emoji account-name color]} {:keys [public-key address  path]}]]
+                   (let [key-uid   (get-in db [:profile/profile :key-uid])
+                         sha3-pwd   (native-module/sha3  (security/safe-unmask-data password))
+                         account-config (clj->js {:key-uid key-uid
```
Comment (ilmotta) [inline src/status_im2/contexts/wallet/events.cljs]: `clj->js` is unnecessary because the effect takes care of calling `utils.transforms/clj->json` on the entire payload before making the RPC call. Same for the other event `:wallet-2/create-derived-addresses`.

**Review by ilmotta — APPROVED** [review]
I didn't review the UI part @J-Son89, and focused only on the re-frame part as you requested. I left just minor comments 🚀

File: src/status_im2/common/standard_authentication/standard_auth/view.cljs [hunk 15]
```diff
@@ -1,14 +1,14 @@
...
-    [react-native.core :as rn]))
+   [quo2.core :as quo]
+   [quo2.theme :as quo.theme]
+   [reagent.core :as reagent]
+   [utils.re-frame :as rf]
+   [react-native.touch-id :as biometric]
+   [utils.i18n :as i18n]
+   [taoensso.timbre :as log]
+   [status-im2.common.standard-authentication.enter-password.view :as enter-password]
+   [react-native.core :as rn]))
```
Comment (ulisesmac) [inline src/status_im2/common/standard_authentication/standard_auth/view.cljs]: I think @J-Son89 fixed the formatting here, but our `lint-fix` tool will revert this change.
@ilmotta do we have a way to fix this? maybe in this PR?

File: src/status_im2/contexts/wallet/events.cljs (hunk 12 above)
Comment (ulisesmac) [reply src/status_im2/contexts/wallet/events.cljs]: @J-Son89  @ilmotta I'm not sure about this.
If the order is not important (like right now that th effect returned is just one) I think we could describe the effect at the root of the map returned.

I mean, it's less verbose and the result is the same. We could use the `:fx` syntax:
```clojure
{:fx [ [:effect-1 params-1] [:effect-2 params-2] ,,,]
 ,,,}
```
when we need to describe many effects **AND** the order is important.
WDYT?

File: src/status_im2/common/standard_authentication/standard_auth/view.cljs (hunk 15 above)
Comment (ilmotta) [reply src/status_im2/common/standard_authentication/standard_auth/view.cljs]: @ulisesmac, this issue seems to be related to how `clojure-lsp` cleans up namespaces. It aligns the opening square bracket `[` with the colon in `:require`. The way we configured zprint causes it to align the namespaces with the letter `r`.

When I run `make lint-fix` zprint aligns the namespaces to `r` in `:require`, which seems to be correct.

File: src/status_im2/contexts/wallet/events.cljs (hunk 12 above)
Comment (ilmotta) [reply src/status_im2/contexts/wallet/events.cljs]: But why have two ways to do the same thing @ulisesmac? The old API only exists for backwards compatibility. The recommendation from re-frame is pretty solid and we added it to our guidelines too. All the docs in re-frame refer to the `:fx` effect. If we stick to vanilla re-frame we will be following the official docs people read.

> While event handlers can return a map of arbitrary effects, it is now recommended that they only return a map containing two standard keys :db and :fx.
>
> -- https://github.com/day8/re-frame/blob/39adca93673f334dc751ee2d99d340b51a9cc6db/docs/FAQs/BestPractice.md#use-the-fx-effect

This `now` the re-frame doc refers to is already 3 years old btw.

> when we need to describe many effects AND the order is important.

This puts an additional burden on devs to think about which syntax to use. Order in effects is 99% of the time desirable. Even when we don't need, some time in another PR, a developer might need, and they would need to refactor the code to use the `:fx` effect. Obviously devs will probably forget about this, so it's better to just stick to the best practice.

File: src/status_im2/common/standard_authentication/standard_auth/view.cljs (hunk 15 above)
Comment (ulisesmac) [reply src/status_im2/common/standard_authentication/standard_auth/view.cljs]: > When I run make lint-fix zprint aligns the namespaces to r in :require, which seems to be correct.

I think this behavior is not correct. it is a list, not a function call:
```clojure
;; list:
(:require
 :b 
 :c
 [])
 
;; fn call
(require
  :b 
  :c
  []) 
```

File: src/status_im2/contexts/wallet/events.cljs (hunk 12 above)
Comment (ulisesmac) [reply src/status_im2/contexts/wallet/events.cljs]: Got it, makes sense to have uniform code, also stick to best practices, as far as I remembered, returning only `:fx` was just a suggestion, not the best practice, now I see this is different.

(btw, I'm still not happy about how re-frame's API has evolved)

**Review by OmarBasem — APPROVED** [review]

File: src/status_im2/common/standard_authentication/standard_auth/view.cljs (hunk 15 above)
Comment (J-Son89) [reply src/status_im2/common/standard_authentication/standard_auth/view.cljs]: apolgies, it's probably my local formatter but not the linter. I'll check

**Review by smohamedjavid — APPROVED** [review]
Nice work! 🚀

**Comment by J-Son89** [comment]
@ilmotta - I took a look at that integrations test file. I realised it is a bit messy as many tests mixed in together and that won't scale. I have started another pr to separate that file out and I am going to add the new integration tests in that pr 👍

File: src/status_im2/common/standard_authentication/standard_auth/view.cljs (hunk 15 above)
Comment (ilmotta) [reply src/status_im2/common/standard_authentication/standard_auth/view.cljs]: > I think this behavior is not correct. it is a list, not a function call:

I meant correct from zprint's point of view. But I agree how clojure-lsp formats is the correct way. @yqrashawn has the greatest knowledge about zprint.

@yqrashawn, do you know if we can tell zprint to format namespaces identically to clojure-lsp? At the moment, `clojure-lsp clean-ns` command has one less whitespace than zprint.

File: src/status_im2/contexts/wallet/events.cljs (hunk 12 above)
Comment (ilmotta) [reply src/status_im2/contexts/wallet/events.cljs]: > (btw, I'm still not happy about how re-frame's API has evolved)

I started to appreciate the API more after reading this long issue and all the insightful comments. Check out issue 639 in https://github.com/day8/re-frame/issues @ulisesmac

**Comment by ilmotta** [comment]
> @ilmotta - I took a look at that integrations test file. I realised it is a bit messy as many tests mixed in together and that won't scale. I have started another pr to separate that file out and I am going to add the new integration tests in that pr 👍

Awesome @J-Son89! I remember past discussions where people mentioned that file could get some love and be split up into different contexts/domains.

In terms of test reliability, these tests fail sometimes to me and then I need to re-run them, which is expected because we're hitting real stuff, but for the most part it's fine. Just a reminder, because I don't know how reliable the RPC endpoints are for the wallet domain.

**Comment by cammellos** [comment]
> > @ilmotta - I took a look at that integrations test file. I realised it is a bit messy as many tests mixed in together and that won't scale. I have started another pr to separate that file out and I am going to add the new integration tests in that pr 👍
> 
> Awesome @J-Son89! I remember past discussions where people mentioned that file could get some love and be split up into different contexts/domains.
> 
> In terms of test reliability, these tests fail sometimes to me and then I need to re-run them, which is expected because we're hitting real stuff, but for the most part it's fine. Just a reminder, because I don't know how reliable the RPC endpoints are for the wallet domain.

I don't think they should fail, I don't believe any of them relies on network requests, so if they are failing it might be that we are doing something wrong or actually rely on some network conditions (in status-go, or in the tests themselves, or there's some actual bug). Would you mind reporting the issue next time they fail with logs?
Thanks

**Comment by cammellos** [comment]
> @ilmotta - I took a look at that integrations test file. I realised it is a bit messy as many tests mixed in together and that won't scale. I have started another pr to separate that file out and I am going to add the new integration tests in that pr 👍

One thing to be aware of, is that these tests are fairly slow, so splitting them out is probably ok, but we should probably find a way to create as little accounts as possible, which would couple the tests unfortunately, but otherwise they'd likely start taking a lot of time, just something to consider.

**Comment by ilmotta** [comment]
> > > @ilmotta - I took a look at that integrations test file. I realised it is a bit messy as many tests mixed in together and that won't scale. I have started another pr to separate that file out and I am going to add the new integration tests in that pr 👍
> > 
> > 
> > Awesome @J-Son89! I remember past discussions where people mentioned that file could get some love and be split up into different contexts/domains.
> > In terms of test reliability, these tests fail sometimes to me and then I need to re-run them, which is expected because we're hitting real stuff, but for the most part it's fine. Just a reminder, because I don't know how reliable the RPC endpoints are for the wallet domain.
> 
> I don't think they should fail, I don't believe any of them relies on network requests, so if they are failing it might be that we are doing something wrong or actually rely on some network conditions (in status-go, or in the tests themselves, or there's some actual bug). Would you mind reporting the issue next time they fail with logs? Thanks

I'll keep an eye @cammellos. I assumed this was known by others. It happened quite a few times to me, locally and in the CI. It's still rare though, so I can't reproduce and I can't find past logs either, they're long gone and I won't be able to find the Jenkins job that failed.

File: src/status_im2/contexts/wallet/events.cljs (hunk 12 above)
Comment (ulisesmac) [reply src/status_im2/contexts/wallet/events.cljs]: OMG, a long issue. I will take a look :+1:  thanks!

File: src/status_im2/common/standard_authentication/standard_auth/view.cljs (hunk 15 above)
Comment (yqrashawn) [reply src/status_im2/common/standard_authentication/standard_auth/view.cljs]: Will look into this maybe later next week https://github.com/status-im/status-mobile/issues/17641

**Review by vkjr — APPROVED** [review]

**Comment by pavloburykh** [comment]
Hey @J-Son89! Thanks for the PR! 

Got a couple of questions: 

1. 

> To test this branch I worked with @mmilad75's pr: https://github.com/status-im/status-mobile/pull/17447 - see the video attached

Why are we not testing this feature in one branch if this PR depends on https://github.com/status-im/status-mobile/pull/17447? 

2. Is it expected that creation of new account through old UI is broken? 

https://github.com/status-im/status-mobile/assets/97245802/da679703-d374-448a-a214-7472d9a06536


3. Is creation of watch only account is in scope of this PR? Currently it is not implemented
photo_2023-10-16 17 39 22

4. Should I pay pay attention at UI issues? for example:

### ISSUE 1 Laggy emoji picker screen

It is twitchy while scrolling and sometimes responses with a delay after selecting emoji


https://github.com/status-im/status-mobile/assets/97245802/13d89920-0327-4e33-b77b-62fbcca8b758


### ISSUE 2 Issues related to slider location on create account screen

**Actual result:** 

On Android slider above the keyboard overlaps other elements (Android 12, Samsung Galaxy A52)

photo_2023-10-16 17 38 42


On IOS it is hidden behind the keyboard (IOS 16.1 iPhone X)

photo_2023-10-16 17 37 40

**Comment by pavloburykh** [comment]
@J-Son89 could you please resolve conflicts in src/status_im2/subs/root.cljs ?

**Comment by J-Son89** [comment]
Thanks @pavloburykh, I just rebased there

wrt your questions:

Issue 1 - that is a seprate work. We're aware that it is a bit laggy but have decided to not be overly concerned on that just yet as we make some more progress with the wallet functionality/ UI. Perhaps in a few weeks we can put more focus on improving that then 👌 


ISSUE 2 - I will address this. Feel free to wait with testing until I address it.

Watch only is not in scope of this pr, the scope of this pr is to add the default new account with most basic settings possible

_Is it expected that creation of new account through old UI is broken?_

This pr should not affect the old UI from working in any way, should be fully functional as before.


_Why are we not testing this feature in one branch if this PR depends on https://github.com/status-im/status-mobile/pull/17447?_
Apologies I should really have removed this info, I added this before I remembered/realised that it is possible to see the list of accounts added in the old UI.
I.e I can add an account in the new ui and then see them displayed in the old UI. albeit I can not see the emoji, color used etc. However I will add those details as a follow up once this pr and Milad's pr is merged in

**Comment by pavloburykh** [comment]
@J-Son89 thank you for your answeres!

>Is it expected that creation of new account through old UI is broken?

>This pr should not affect the old UI from working in any way, should be fully functional as before.


I cannot create a new account via old UI if I have already added an account via new UI. 


**Steps:** 

1. Add account via new UI
2. Relogin
3. Try to add one more account via old UI

**Actual result:** Add account button does not respond

Perhaps such flow is not valid, but anyway the behaviour if weird. 

https://github.com/status-im/status-mobile/assets/97245802/1cc5aade-35d1-4ded-bc43-f09a9bbb00fc

**Comment by J-Son89** [comment]
Interesting, I had not even considered that as an option. 

Let me look into that although I believe that we can consider these separate. I.e when we release the new wallet the user will not have access to the old wallet anymore. 

However it does beg the question that perhaps we should remove the new wallet from the public releases until it is very stable.

**Comment by J-Son89** [comment]
I will verify if that flow is even "possible" 👌

**Comment by J-Son89** [comment]
@pavloburykh - can we leave the android issue 2 as a follow up? 
It is quite UI related and I have another pr which adds some code which I will use there for this ->
https://github.com/status-im/status-mobile/pull/17672
Once 17672 is merged in I will add a follow up to address that issue on create account page.
Wdyt? 🤔

**Comment by pavloburykh** [comment]
> Once 17672 is merged in I will add a follow up to address that issue on create account page.
> Wdyt? 🤔

@J-Son89 SGTM. I guess we still need quick design-review. cc @Francesca-G Please mind the above known issues during review. Thanks!

**Comment by J-Son89** [comment]
@pavloburykh, @Francesca-G - we are still somewhat not requiring design reviews on wallet screens. If the developer wants them that is fine but imo I do not need one here as I intend to clean this UI at a later point anyway.
The reason for this is not to mispend @Francesca-G's time (or the developer) when a lot of changes are going to be made and the final polish is not ready to happen. We will be fighting small paddings/margins a lot otherwise.
Soon enough we will stabilise this and make it mandatory similar to the rest of the application ofc :)

**Comment by J-Son89** [comment]
@pavloburykh this pr is working ok? 
This pr has adjustments made on Status Go. As I understand it these changes need to be merged on Status Go now, and then I will update this branch to point to lastest on Status go and we do a last check.

cc @cammellos can I merge that pr on Status go? 🤔

**Comment by pavloburykh** [comment]
> @pavloburykh this pr is working ok?

Hey @J-Son89! It was working okay when I moved it to merge column last week. But now go  and mobile branches are outdated. That's why we need to rebase both branches and perform another round of testing (including re-run e2e) before merge to make sure no issues introduced by updated go.

**Comment by J-Son89** [comment]
Thanks @pavloburykh. The changes needed on Status Go were merged and I have updated this pr to point at the right version of Status Go.
I tested this pr locally and in the release build and it's working as expected. Can you take another look before I look to merge it?

**Comment by pavloburykh** [comment]
> I tested this pr locally and in the release build and it's working as expected. Can you take another look before I look to merge it?

Thank you @J-Son89! Let's have another e2e run before merging.

**Comment by J-Son89** [comment]
perfect! 👌

**Comment by pavloburykh** [comment]
@J-Son89 hold on with merging please. E2E results are suspicious, I will check tomorrow.

**Comment by pavloburykh** [comment]
@J-Son89 PR is ready for merge.

**Comment by pavloburykh** [comment]
> Once 17672 is merged in I will add a follow up to address that issue on create account page.
> Wdyt? 🤔

Hey @J-Son89! Could you please clarify if the above ISSUE 1 and 2 were logged as followups as I couldn't find them. If not  please let me know and I will log them.

**Comment by J-Son89** [comment]
Hi @pavloburykh - apologies I assumed - ISSUE 1 Laggy emoji picker screen -had a corresponding issue as it is something Javid and I had discussed. I believe he is on holiday this week and I can not find the issue. I will create one anyway for tracking and send it to Javid, he can close the issue if it's a duplicate. 👍 

with 2 - it's a more general problem and once we have the code in Ulises & my pr - https://github.com/status-im/status-mobile/pull/17737
we will then be using this in the various pages of the wallet. I didn't create an issue for this specific page as it's a generic problem in the wallet screens which we intend to address.
is that ok? 🤔

**Comment by J-Son89** [comment]
@pavloburykh - https://github.com/status-im/status-mobile/issues/17889
btw, we will not be addressing the emoji picker performance until 1.27 requirements have been met, I think it's a nice to have and not that much of a blocker. wdyt?

**Comment by mariia-skrypnyk** [comment]
[Here](https://github.com/status-im/status-mobile/issues/18841) is a followup of **ISSUE 2**
